From be4f67782f99945f5986f7ac9b0da9989da9e932 Mon Sep 17 00:00:00 2001
From: Tendai Makumire <tendai.makumire@imd-tec.com>
Date: Thu, 9 Apr 2026 05:01:23 +0100
Subject: [PATCH] usb: typec: ucsi: Add API to get type-c connector orientation
 from PPM

UCSI v1.1.0 does not define a way to get the current
connector orientation when a port partner
is connected. However, this is a feature which
may be supported through other registers
in the PPM. So we provide an optional
API for it so that the type-c orientation
can be retrieved and used in the UCSI
core.

Note - Only header file changes taken from
common-kernel to allow msm-kernel to compile
---
 drivers/usb/typec/ucsi/ucsi.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/typec/ucsi/ucsi.h b/drivers/usb/typec/ucsi/ucsi.h
index 192728dc8..346f1e04b 100644
--- a/drivers/usb/typec/ucsi/ucsi.h
+++ b/drivers/usb/typec/ucsi/ucsi.h
@@ -40,6 +40,7 @@ struct ucsi_altmode;
  * @sync_write: Blocking write operation
  * @async_write: Non-blocking write operation
  * @update_altmodes: Squashes duplicate DP altmodes
+ * @get_conn_orientation: Get type-c connector orientation
  *
  * Read and write routines for UCSI interface. @sync_write must wait for the
  * Command Completion Event from the PPM before returning, and @async_write must
@@ -54,6 +55,7 @@ struct ucsi_operations {
 			   const void *val, size_t val_len);
 	bool (*update_altmodes)(struct ucsi *ucsi, struct ucsi_altmode *orig,
 				struct ucsi_altmode *updated);
+	enum typec_orientation (*get_conn_orientation)(struct ucsi *ucsi, u8 num);
 };
 
 struct ucsi *ucsi_create(struct device *dev, const struct ucsi_operations *ops);
-- 
2.41.0

